indexing-service/src/main/java/io/druid/indexing/common/task/HadoopIndexTask.java

```diff
@@ -149,6 +151,10 @@ public String getClasspathPrefix()
   @Override
   public TaskStatus run(TaskToolbox toolbox) throws Exception
   {
+    //Setting up UsedSegmentLister which is used by DatasourcePathSpec to get the list of used
+    //segments
+    DatasourcePathSpec.setUsedSegmentLister(new OverlordActionBasedUsedSegmentLister(toolbox));
```

Comment:
@gian main issue is that we want different behavior. with standalone we want the worker process to directly access metadata store, but with task we want it to access via overlord action.
It would've been fine if guice injection worked and I could configure things in CliPeon and CliInternalHadoopIndexer but that does not work because HadoopDruidIndexerConfig has its own guice Injector that does not know about the modules in CliXXX classes.